ql/src/java/org/apache/hadoop/hive/ql/exec/HashTableSinkOperator.java

```diff
@@ -179,18 +178,18 @@ protected void initializeOp(Configuration hconf) throws HiveException {
     }
     try {
       TableDesc keyTableDesc = conf.getKeyTblDesc();
-      AbstractSerDe keySerde = (AbstractSerDe) ReflectionUtils.newInstance(keyTableDesc.getDeserializerClass(),
+      AbstractSerDe keySerde = (AbstractSerDe) ReflectionUtils.newInstance(keyTableDesc.getSerDeClass(),
```

Comment:
Nit: please call this keySerDe for consistency.

Reply:
Done.